### ISSUE TITLE
Schematron currently checks if the content in <agencyCode> is compliant with ISO 15511 even if @repositoryEncoding is not set

### ISSUE THREAD

**Issue by kerstarno** [SAA-SDT/eas-schematrons#10]
## Creator of issue

1. Kerstin Arnold
2. EAD team lead, TS-EAS
3. @kerstarno
4. kerstin.arnold@archivesportaleurope.net

## The issue relates to

- [ ] EAC-CPF schema issue
- [ ] EAC-CPF Tag Library issue
- [x] EAD schema issue
- [ ] EAD Tag Library issue
- [ ] Schema issue
- [ ] Tag Library issue
- [x] Suggestions for all schemas
- [ ] Suggestions for all Tag Libraries
- [ ] Other

## Wanted change/feature

* Text: This might be a similar remnant from EAD3 Schematron rules as requiring ISO 8601 compliant dates even if `@dateEncoding` isn't set (see #7). At the moment, in both the XSD and the RNG (when used with Schematron), there seems to be a check for the content of `<agencyCode>` to be compliant with ISO 15511, so that the following error message is shown even if `@repositoryEncoding` isn't set (yet) in `<control>`: "If the repositoryEncoding is set to ISO 15511, then the format of the value of the agencyCode element is constrained to that of the International Standard Identifier for Libraries and Related Organizations (ISIL: ISO 15511): a prefix, a hyphen, and an identifier." 
* This message should only be shown, if `@repositoryEncoding` is present and is indeed used with the value "iso15511".

**Comment by fordmadox** [comment]
That was the decision made for the EAC CPF 2.0 release (ditto for the default of ISO 8601, which was also present in EAD3).  I will update the 'ead4' branch for this decision, but since it differs from the EAC decision, I'll also add Marie to this ticket for future review.

**Comment by fordmadox** [comment]
Same for script and country codes, since those lists were always the de-facto defaults, and so, didn't need to be specified.  

But for this issue, I'm just updating the repositoryEncoding test.

**Comment by fordmadox** [comment]
See https://github.com/SAA-SDT/eas-schematrons/pull/13/files

**Comment by kerstarno** [comment]
Re-tested with the XSD and the RNG (with the "ead4" branch Schematron) and can confirm that the check is now only done when `@repositoryEncoding` is used/present in `<control>` and has the value "iso15511". 

Will admit that I didn't remember that decision from the EAC-CPF revision anymore. I think, what mainly made me wonder was the fact that the messaged starts with "If the repositoryEncoding is set to ISO 15511...". As long as I haven't used the `@repositoryEncoding` attribute to start with, that message seems off. The same for `@countryCode`, `@scriptCode`, and `@scriptOfElement`. As long as I don't have the respective `@...Encoding` attributes added to `<control>`, the error message can't - to my mind - require the value of these attributes to follow a specific ISO code (or EAS list). The first message to show in that case should - again, to my mind - rather be something like "Please add `@...Encoding` to `<control>` and specify the code list that the values of `@[name of the attribute in question]` are expected to follow."

**Comment by fordmadox** [comment]
Agreed.  I think that all the Schematron messages should be reviewed and updated!!!  Most were just added as a quick example.
